src/CSharp/CodeCracker/Usage/MethodAnalyzers/MethodChecker.cs

```diff
@@ -0,0 +1,106 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace CodeCracker.Usage.MethodAnalyzers
+{
+    public class MethodChecker
+    {
+        private readonly SyntaxNodeAnalysisContext context;
+        private readonly DiagnosticDescriptor diagnosticDescriptor;
+
+        public MethodChecker(SyntaxNodeAnalysisContext context, DiagnosticDescriptor diagnosticDescriptor)
+        {
+            this.context = context;
+            this.diagnosticDescriptor = diagnosticDescriptor;
+        }
+
+        public void AnalyzeConstrutor(MethodInformation methodInformation)
+        {
+            if (ConstructorNameNotFound(methodInformation) || MethodFullNameNotFound(methodInformation.MethodFullDefinition))
+            {
+                return;
+            }
+            var argumentList = ((ObjectCreationExpressionSyntax)context.Node).ArgumentList;
+            var arguments = GetArguments(argumentList);
+            Execute(methodInformation, arguments, argumentList);
+        }
+
+        private bool ConstructorNameNotFound(MethodInformation methodInformation)
+        {
+            return AbreviatedConstructorNameNotFound(methodInformation) && QualifiedConstructorNameNotFound(methodInformation);
+        }
+
+        private bool AbreviatedConstructorNameNotFound(MethodInformation methodInformation)
+        {
+            var objectCreationExpressionSyntax = (ObjectCreationExpressionSyntax) context.Node;
+            var identifier = objectCreationExpressionSyntax.Type as IdentifierNameSyntax;
+            return identifier?.Identifier.ValueText != methodInformation.MethodName;
+        }
+
+        private bool QualifiedConstructorNameNotFound(MethodInformation methodInformation)
+        {
+            var objectCreationExpressionSyntax = (ObjectCreationExpressionSyntax)context.Node;
+            var identifier = objectCreationExpressionSyntax.Type as QualifiedNameSyntax;
+            return identifier?.Right.ToString() != methodInformation.MethodName;
```

Comment:
@andrecarlucci You are comparing by string, not by symbol. This probably can cause unnexpected mnatches. If I name something `Uri`, it will probably match on the `AbreviatedConstructorNameNotFound` method, right? This will cause incorrect diagnostics. Can we do better?